pandas/core/internals.py

```diff
@@ -1455,15 +1473,42 @@ def should_store(self, value):
         return com.is_integer_dtype(value) and value.dtype == self.dtype
 
 
-class TimeDeltaBlock(IntBlock):
+class DatetimeLikeBlockMixin(object):
+
+    @property
+    def _na_value(self):
+        return tslib.NaT
+
+    @property
+    def fill_value(self):
+        return tslib.iNaT
+
+    def _try_operate(self, values):
+        """ return a version to operate on """
+        return values.view('i8')
+
+    def get_values(self, dtype=None):
+        # return object dtype as boxed values, such as Timestamps/Timedelta
+        if dtype == object:
+            return lib.map_infer(self.values.ravel(),
+                                 self._box_func).reshape(self.values.shape)
+        return self.values
+
+    def quantile(self, values, qs, **kwargs):
+        values = values.view('i8')
+        result = Block.quantile(self, values, qs, **kwargs)
```

Comment:
Correct. Considering NaN handling (also ``nanpercentile``), it is better to call ``quantile`` per columns ATM. 

I found other issues related to quantile (#12772), and changes gets larger than I thought.